src/backend/src/signer.rs

```diff
@@ -0,0 +1,60 @@
+//! Code for inetracting with the chain fusion signer.
+use crate::state::{CYCLES_LEDGER, SIGNER};
+use candid::{CandidType, Deserialize, Nat, Principal};
+use ic_cycles_ledger_client::{Account, ApproveArgs, ApproveError, Service as CyclesLedgerService};
+use ic_ledger_types::Subaccount;
+use serde_bytes::ByteBuf;
+
+#[derive(CandidType, Deserialize, Debug, Clone, Eq, PartialEq)]
+pub enum AllowSigningError {
+    Other(String),
+    FailedToContactCyclesLedger,
+    ApproveError(ApproveError),
+}
+
+/// Enables the user to sign transactions.
+///
+/// Signing costs cycles.  Managing that cycle payment can be painful so we take care of that.
+pub async fn allow_signing() -> Result<(), AllowSigningError> {
+    let cycles_ledger: Principal = *CYCLES_LEDGER;
+    let signer: Principal = *SIGNER;
+    let caller = ic_cdk::caller();
+    let expected_fee = 1_000_000_000u64;
```

Comment:
Can we create it as a const and add a reference to why is it `1_000_000_000`?

Or is this even the fee or the units of TCycles?

Reply:
Done